chia/consensus/block_header_validation.py

```diff
@@ -491,6 +496,28 @@ def validate_unfinished_header_block(
     if q_str is None:
         return None, ValidationError(Err.INVALID_POSPACE)
 
+    # 5c. Check plot id is not present within last `NUM_DISTINCT_CONSECUTIVE_PLOT_IDS` blocks.
+    if height >= constants.SOFT_FORK3_HEIGHT:
+        curr_optional_block_record: Optional[BlockRecord] = prev_b
+        plot_id = get_plot_id(header_block.reward_chain_block.proof_of_space)
+        curr_sp = cc_sp_hash
+        sp_count = 1
+
+        while curr_optional_block_record is not None and sp_count < constants.UNIQUE_PLOTS_WINDOW:
+            prefix_bits = calculate_prefix_bits(constants, curr_optional_block_record.height)
+            if passes_plot_filter(
+                prefix_bits,
+                plot_id,
+                curr_optional_block_record.pos_ss_cc_challenge_hash,
+                curr_optional_block_record.cc_sp_hash,
+            ):
+                return None, ValidationError(Err.INVALID_POSPACE)
+
+            if curr_optional_block_record.cc_sp_hash != curr_sp:
+                sp_count += 1
```

Comment:
maybe this is a nit, but this is where `sp_count` will be `>= constants.UNIQUE_PLOTS_WINDOW`. In which case it's time to end the loop. Yet, we still look up one more block record before we exit.